### ISSUE TITLE
Crash in view-change

### ISSUE THREAD

**Issue by ineiti** [dedis/cothority#1691]
Here is a crash-report I got from running the conodes:

```
2 19/02/02 23:36:00.586615035: (    byzcoin.(*Service).sendViewChangeReq: 120) - tls://pop.dedis.ch:7778 sending view-change request for view: ID: 5c32d79c59b4419b1929fd2a28b78ff1b78bc64207da374ca48373ebce193f56, Gen: aaa5b2db9523c5c66eedac22d878bc3d718d19f60d944004a53a9b977ed88ba3, Leader: 4
panic: runtime error: index out of range

goroutine 8 [running]:
github.com/dedis/cothority/byzcoin.(*Service).sendViewChangeReq(0xc420108dc0, 0xc420285824, 0x20, 0x20, 0xc42018b740, 0x20, 0x20, 0x4, 0x0, 0x0)
        /Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange.go:126 +0x4a8
github.com/dedis/cothority/byzcoin.(*Service).(github.com/dedis/cothority/byzcoin.sendViewChangeReq)-fm(0xc420285824, 0x20, 0x20, 0xc42018b740, 0x20, 0x20, 0x4, 0x20, 0x20)
        /Users/ligasser/src/github.com/dedis/cothority/byzcoin/service.go:1255 +0x51
github.com/dedis/cothority/byzcoin/viewchange.(*Controller).processAnomaly(0xc42008c5f0, 0xc420285824, 0x20, 0x20, 0xc42018b740, 0x20, 0x20, 0x4, 0x1f5b187e657efbe3, 0x6ebcec998023d195, ...)
        /Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange/viewchange.go:238 +0x15e
github.com/dedis/cothority/byzcoin/viewchange.(*Controller).Start(0xc42008c5f0, 0x1f5b187e657efbe3, 0x6ebcec998023d195, 0xc42018b740, 0x20, 0x20, 0xba43b7400, 0x1)
        /Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange/viewchange.go:217 +0xbbe
created by github.com/dedis/cothority/byzcoin.(*viewChangeManager).start
        /Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange.go:53 +0x16d
```

and another one:

```
2 19/29/01 18:53:36.554783495: (          byzcoin.(*Service).sendNewView: 155) - tls://pop.dedis.ch:7772 sending new-view request for view: ID: ab60e2c091375db0b1d47c4862fbc7e9ce38129135feb960855bba63d6f9268a, Gen: aaa5b2db9523c5c66eedac22d878bc3d718d19f60d944004a53a9b977ed88ba3, Leader: 2
E 19/29/01 18:53:42.511814855: (    onet.(*Overlay).CreateProtocol.func1: 557) - viewchange_ftcosi.Dispatch() created in service ByzCoin returned error timeout, did you forget to call Start?
2 19/29/01 18:53:44.467054678: (                          runtime.call32: 573) - tls://pop.dedis.ch:7772 finished view-change ftcosi
panic: send on closed channel

goroutine 18796 [running]:
github.com/dedis/cothority/ftcosi/protocol.(*FtCosi).Start(0xc4206c0a90, 0x3fd5555555555555, 0x3ff965fea53d6e3c)
	/Users/ligasser/src/github.com/dedis/cothority/ftcosi/protocol/protocol.go:318 +0x177
github.com/dedis/cothority/byzcoin.(*Service).startViewChangeCosi(0xc420130f20, 0xf85661fa2772c4c9, 0x414c37fbfaea2baf, 0xc420433ec0, 0x4, 0x4, 0xb5f140, 0xc4200cf290, 0xc4206bf0e0, 0x3, ...)
	/Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange.go:282 +0x53d
github.com/dedis/cothority/byzcoin.(*Service).sendNewView.func1(0xc420130f20, 0xf85661fa2772c4c9, 0x414c37fbfaea2baf, 0xc420433ec0, 0x4, 0x4, 0xb5f140, 0xc4200cf290, 0xc4206bf0e0, 0x3, ...)
	/Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange.go:175 +0xc3
created by github.com/dedis/cothority/byzcoin.(*Service).sendNewView
	/Users/ligasser/src/github.com/dedis/cothority/byzcoin/viewchange.go:170 +0x316
```

The code is from the following branch:

https://github.com/dedis/cothority/tree/personhood.online

**Comment by Gilthoniel** [comment]
First issue is fixed since #1781

**Comment by Gilthoniel** [comment]
For the second issue, it looks like the view change is using the previous CoSi protocol so I'm closing this issue.
